src/main/java/frc/robot/RobotContainer.java

```diff
@@ -73,11 +81,109 @@ public class RobotContainer {
   Command driveFieldOrientedAngularVelocitySim =
       drivebase.driveFieldOriented(driveAngularVelocitySim);
 
+  private final Command rumble(double power, double duration) {
+    return Commands.runOnce(
+            () -> {
+              driverXbox.getHID().setRumble(RumbleType.kBothRumble, power);
+              codriverXbox.getHID().setRumble(RumbleType.kBothRumble, power);
+            })
+        .andThen(Commands.waitSeconds(duration))
+        .andThen(
+            () -> {
+              driverXbox.getHID().setRumble(RumbleType.kBothRumble, OperatorConstants.RUMBLE_REST);
+              codriverXbox
+                  .getHID()
+                  .setRumble(RumbleType.kBothRumble, OperatorConstants.RUMBLE_REST);
+            });
+  }
+
+  private final Command autoOuttakeCoral() {
+    return Commands.sequence(
```

Comment:
```suggestion
  private final Command autoOuttakeCoral = Commands.sequence(
```

Alot of these don't need to be methods. You can just make them variables.